google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -371,19 +377,37 @@ def query_options
         #     puts "User #{row[:id]} is #{row[:name]}"
         #   end
         #
+        # @example Query using tag for request query statistics collection.
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #
+        #   db = spanner.client "my-instance", "my-database"
+        #
+        #   request_options = { tag: "Read-Users" }
+        #   results = db.execute_query "SELECT * FROM users",
+        #                              request_options: request_options
+        #
+        #   results.rows.each do |row|
+        #     puts "User #{row[:id]} is #{row[:name]}"
+        #   end
+        #
         def execute_query sql, params: nil, types: nil, single_use: nil,
-                          query_options: nil, call_options: nil
+                          query_options: nil, request_options: nil,
```

Comment:
Just realised that we added the `request_options` as the sixth parameter, where that position used to be used by the `call_options`. The type of both parameters is a map, which might be silently fail for users who had previous code written and were not naming the arguments.

Do you think it should be worth to move it to be the last argument of the method?
Also, could you make sure that this PR is a breaking change, since it updates this interface? (You can do that through the commit message, see https://www.conventionalcommits.org/en/v1.0.0/#specification BREAKING CHANGE).

Reply:
These are named arguments with the default value. In Ruby named arguments are like hash so, the order has no impact while calling.  Here `call_options` are for the GAPIC GRPC client so moved to last, to separate from spanner arguments parameters.

Reply:
Makes sense, thanks for the explanation